entc/gen/graph.go

```diff
@@ -55,6 +55,24 @@ type (
 
 		// Hooks holds an optional list of Hooks to apply on the graph before/after the code-generation.
 		Hooks []Hook
+
+		// Annotations that are injected to the Config object can be accessed
+		// globally in all templates. In order to access an annotation from a
+		// graph template, do the following:
+		//
+		//	{{- with $.Annotations.GQL }}
+		//		{{/* Annotation usage goes here. */}}
+		//	{{- end }}
+		//
+		// For type templates, we access the Config field to access the global
+		// annotations, and not the type-specific annotation.
+		//
+		//	{{- with $.Config.Annotations.GQL }}
+		//		{{/* Annotation usage goes here. */}}
+		//	{{- end }}
+		//
+		// Note that the mapping is from the annotation-name (e.g. "GQL") to a JSON decoded object.
+		Annotations Annotations
```

Comment:
@alexsn - see the docs on how to access these annotations in templates.

I think the next step will be to add a `gen.Extension` (or a `gen.Plugin`), that will wrap the codegen hooks, annotations, and templates in one option.